### ISSUE TITLE
Anonymous sharing

### ISSUE THREAD

**Issue by jayrbolton** [jayrbolton/metadat#6]
This may be way out of scope or unnecessary, but wanted to leave this here...

Users should be able to share a dat anonymously. Other users who download this dat would not be able to trace (or easily trace) the source device of the dat share. Could possibly do some kind of onion routing style thing over dat

Maybe related:
https://www.net.t-labs.tu-berlin.de/papers/M-UORWP2PA-08.pdf
https://www.freehaven.net/anonbib/cache/conf/iwsec/PalmieriP14.pdf

**Comment by karissa** [comment]
Dat already does this so that third parties don't know but not clear if that's also the case for those who have the link

**Comment by jayrbolton** [comment]
Yeah I was thinking in terms of being able to share a dat link with someone, but they wouldn't be able to trace any IP of who is sharing the dat. I was thinking specifically about whistleblowing and things like SecureDrop. I think it is possible to do an onion routing thing over P2P. This could def be a separate lib, but part of the DatDrive UI, or whatever we're calling it.

If user A wants to send something to user B, and user A trusts user B, user A could simply encrypt the dat address using user B's pgp pubkey before sending the address, or just send the address directly over encrypted p2p. 

But if user A wants to secure their anonymity and doesn't necessarily trust user B, then they will want to share and seed their data with user B using some kind of proxying. Another way would be for a third-party org to provide dat hosting, where the org's server doesn't save any info on the user and can't read the data in the node. But it would be cooler if this anonymity was built onto the underlying protocal itself.
